include/nuttx/sensors/sensor.h

```diff
@@ -595,6 +603,15 @@ struct sensor_cap           /* Type: Capacitance */
   int32_t rawdata[4];       /* in SI units pF */
 };
 
+struct sensor_gas           /* Type: Gas */
+{
+  uint64_t timestamp;       /* Units is microseconds */
+  float temperature;        /* Temperature in degrees Celsius */
+  float pressure;           /* Pressure in millibar or hPa */
+  float humidity;           /* Relative humidity in rH */
```

Comment:
I did not know how to do that. I thought there has to be a struct associated with the type of sensor, as declared in the g_sensor_info array. Also, using 2 structs wouldn t mean having the timestamp twice? I would appreciate it if you could give me a hint as to how i could reuse sensor_baro and sensor_humi.

Reply:
Hi @simonatoaca nice work, thank you! I think @xiaoxiang781216 wants you use the new sensor subsystem, please take a look at wtgahrs2.c for reference, it is also a multi-sensor chip.

Reply:
@simonatoaca already use the new sensor subsystem.

Reply:
> I did not know how to do that.

you can expose multiple topics from one sensor driver. For example, https://github.com/apache/nuttx/blob/master/drivers/sensors/wtgahrs2.c registers 5 topics.

> I thought there has to be a struct associated with the type of sensor, as declared in the g_sensor_info array.

Yes, but it's better to make one topic just contain one physical measurement. For example, IMU should expose three topics(gyro, accel and mag), instead one topic with all value.

> Also, using 2 structs wouldn t mean having the timestamp twice?

Not necessary, you can get timestamp once, and reuse it for all push_event.

> I would appreciate it if you could give me a hint as to how i could reuse sensor_baro and sensor_humi.

From https://www.bosch-sensortec.com/products/environmental-sensors/gas-sensors/bme680/, BME680 can measure.gas, humidity and pressure individually, so it's better to let driver report three topic(gas, humidity and pressure). Temperature is a special one, since many measurement requires temperature to calibrate the result, so it's contained in many sensor topic.

Reply:
Thank you for the clarification. I started to work on it and registered the sensors separately. However, i am not sure how to retrieve the data correctly, since every lowerhalf translates to a sensor_<type> file. It seems wrong to open all 3 or 4 files separately to get the data from each one. Is there a better way to do this? I could not find any examples in nuttx-apps. Also, I have a calibration function that should be called only once through an ioctl, but the ioctl is bound to a file descriptor.

Reply:
> Thank you for the clarification. I started to work on it and registered the sensors separately. However, i am not sure how to retrieve the data correctly, since every lowerhalf translates to a sensor_ file. It seems wrong to open all 3 or 4 files separately to get the data from each one. Is there a better way to do this?

to handle multiple file handle, you need poll/select, or libuv(https://github.com/apache/nuttx-apps/tree/master/system/libuv).

> I could not find any examples in nuttx-apps. Also, I have a calibration function that should be called only once through an ioctl, but the ioctl is bound to a file descriptor.

it can be done through one of topics.